Sources/EmbraceConfigInternal/EmbraceConfig.swift

```diff
@@ -15,32 +15,23 @@ public class EmbraceConfig {
     let logger: InternalLogger
     let notificationCenter: NotificationCenter
 
-    let deviceIdUsedDigits = 6
-    var deviceIdHexValue: UInt64 = UInt64.max // defaults to everything disabled
-
-    @ThreadSafe var payload: RemoteConfigPayload = RemoteConfigPayload()
-    let fetcher: RemoteConfigFetcher
-
-    @ThreadSafe private(set) var updating = false
     @ThreadSafe private var lastUpdateTime: TimeInterval = Date(timeIntervalSince1970: 0).timeIntervalSince1970
 
-    public var onUpdate: (() -> Void)?
+    let configurable: EmbraceConfigurable
 
-    public init(options: Options, notificationCenter: NotificationCenter, logger: InternalLogger) {
+    public init(
+        configurable: EmbraceConfigurable,
+        options: Options,
+        notificationCenter: NotificationCenter,
+        logger: InternalLogger
+    ) {
```

Comment:
`EmbraceConfig`  has 2 main functions, hold on to the underlying config implementation, and provide the "updateIfNeeded" logic such that config is only updated based on the `minimumUpdateInterval`.

This feels more like a "Manager" type of class because of this change.

Reply:
I see that this class has the `minimumUpdateInterval` logic but I don't understand why it had to be split into two classes still.

Reply:
The alternative would be to require the object that conforms to `EmbraceConfigurable` to provide this behavior. I felt it would be best if the SDK defined this so that it was consistent across all conformers.

Another piece is the "config did update" notification. I didn't want to push that requirement onto the `EmbraceConfigurable` as well without having an explicit interface for it.